### PR TITLE
Solution and Test for issue #58

### PR DESCRIPTION
**Pull request by yjlim5**
Return a guest list for Max fun

### CODE REVIEW

File: test/58.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const expect = require('chai').expect;
+const solution = require('../solutions/58.js').solution;
+// solution = require('../yourSolution').solution;
+
+describe('return smallest number of coins required to return the change', () => {
+  class Employee {
```
Comment (songz) [inline test/58.js]: import this from solution. Doesn't make sense for Employee to be part of tests.

File: solutions/58.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  emp.staff.forEach((staff) => {
+    staff.staff.forEach((staffstaff) => {
+      solution(staffstaff).forEach((emp) => {
+        funValWithEmp += emp.funVal;
+        funStaffListWithEmp.push(emp);
+      });
+    });
+  });
+
+  if (funValWithoutEmp > funValWithEmp) {
```
Comment (songz) [inline solutions/58.js]: `return funValWithoutEmp > funValWithEmp ? funValWithoutEmp : funValWithEmp;`

File: test/58.js (hunk 1 above)
Comment (songz) [inline test/58.js]: required

Comment (yjlim5) [inline test/58.js]: done.

File: solutions/58.js (hunk 2 above)
Comment (yjlim5) [inline solutions/58.js]: done.

Comment (yjlim5) [inline solutions/58.js]: done.

File: solutions/58.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+}
+
+/**
+ * Given an Employee, return an array of employees that creates the maximum fun.
+ * There is one condition: An Employee cannot be on the guestlist if his direct manager is going to the party.
+ * @param {Employee} emp - an Employee with fun value and a list of his staff
+ * @returns {Employee []]} list - Array of employees that creates the maximum fun
+ */
+
```
Comment (songz) [inline solutions/58.js]: almost there! remove empty line

**Pull request merged**